utils/retrievalAPI/retrieveResponse.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-console */
+import readline from "readline";
+import axios from "axios";
+import { createObjectCsvWriter } from "csv-writer";
+
+const formsUrl = "http://localhost:3000";
+
+function getValue(query: string) {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise<string>((resolve) =>
+    rl.question(query, (ans) => {
+      rl.close();
+      resolve(ans);
+    })
+  );
+}
+
+interface formResponse {
+  formID: string;
+  submissionID: string;
+  formSubmission: string;
+}
+
+async function getResponses(
+  formID: string,
+  temporaryToken: string
+): Promise<{ responses: Array<formResponse> }> {
+  try {
+    const formResponses = await axios({
+      method: "GET",
+      url: `${formsUrl}/api/id/${formID}/retrieval`,
+      headers: {
+        authorization: `Bearer ${temporaryToken}`,
+      },
+      timeout: 20000,
+    });
+    return formResponses.data;
+  } catch (e) {
+    console.log("Error retrieving responses");
+    throw e;
+  }
+}
+
+async function saveResponses(responses: Array<formResponse>) {
+  const qIds = Object.keys(JSON.parse(responses[0].formSubmission)).map((q, index) => ({
+    id: q,
+    title: `question ${index}`,
+  }));
+  const csvWriter = createObjectCsvWriter({
+    path: "responses.csv",
+    header: qIds,
+  });
+
+  await csvWriter.writeRecords(responses.map((response) => JSON.parse(response.formSubmission)));
+}
+
+async function confirmResponses(responses: Array<string>, formID: string, temporaryToken: string) {
+  try {
+    await axios({
+      method: "delete",
+      url: `${formsUrl}/api/id/${formID}/retrieval`,
+      headers: {
+        authorization: `Bearer ${temporaryToken}`,
+      },
+      data: responses,
+      timeout: 20000,
+    });
+  } catch (e) {
+    console.log("Error retrieving responses");
+    throw e;
+  }
+}
+
+const main = async () => {
+  try {
+    const email = await getValue("Email address to authenticate with:");
+    const formID = await getValue("Form ID of responses to retrieve:");
```

Comment:
Think it might be worth adding a prompt for the user to accept that in so doing it will remove the responses from our system. Realistically we are the only ones that will use this however, this is open source and so potentially a team working with a form on production could point this script there and use it

Reply:
Good idea.  Added prompt to confirm.